iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/DanglingLineAdderImpl.java

```diff
@@ -148,7 +147,8 @@ public DanglingLineAdder setUcteXnodeCode(String ucteXnodeCode) {
 
     @Override
     public GenerationAdder newGeneration() {
-        return new GenerationAdderImpl();
+        generationAdder = new GenerationAdderImpl();
+        return generationAdder;
```

Comment:
not sure this is correct: I would write
```
return new GenerationAdderImpl(this);
```
And in the `GenerationAdder::add` method, I would set the generationAdder of the parent. I don't know if it changes something, but it's seems better to me that the DanglingLineAdder is modified only once the GenerationAdder has been filled.

Reply:
Regarding what you did for three windings transformers, it's coherent

Reply:
In deed, set the generationAdder in `GenerationAdder::add`, after validity checks, seems a better solution.
DanglingLineAdder will be set only with a GenerationAdder well filled